WearSpeakerSample/wear/src/main/java/com/example/android/wearable/speaker/SpeakerPlayerScreen.kt

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.android.wearable.speaker
+
+import androidx.activity.compose.ReportDrawnAfter
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.media3.common.Player
+import com.google.android.horologist.annotations.ExperimentalHorologistApi
+import com.google.android.horologist.audio.ui.VolumeViewModel
+import com.google.android.horologist.audio.ui.components.actions.SetVolumeButton
+import com.google.android.horologist.media.data.repository.PlayerRepositoryImpl
+import com.google.android.horologist.media.ui.components.PodcastControlButtons
+import com.google.android.horologist.media.ui.screens.player.DefaultMediaInfoDisplay
+import com.google.android.horologist.media.ui.screens.player.PlayerScreen
+import com.google.android.horologist.media.ui.state.PlayerUiController
+import com.google.android.horologist.media.ui.state.PlayerUiState
+import java.io.File
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.flow.first
+
+@OptIn(ExperimentalHorologistApi::class)
+@Composable
+fun SpeakerPlayerScreen(
+    player: Player,
+    volumeViewModel: VolumeViewModel = viewModel(factory = VolumeViewModel.Factory),
+    onVolumeClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val context = LocalContext.current
+    val activity = context.findActivity()
+
+    val volumeUiState by volumeViewModel.volumeUiState.collectAsStateWithLifecycle()
+
+    val outputFileName = "audiorecord.opus"
+    val audioFile = File(activity.filesDir, outputFileName)
+    val audioFileUri = activity.filesDir.path + "/" + outputFileName
+    val playerViewModel = SpeakerPlayerViewModel(
```

Comment:
You can't create view models like this.

If you just need it from the VM, the view model can use the application context.

Reply:
What about using a factory? https://github.com/android/wear-os-samples/pull/1094/files#diff-02cbeefda5435ff1dca10c8c0961531257725dfe73a1caa6227fafb62ea7b40bR61

Reply:
Yep, but what about just passing in filename? Do the filesDir lookup in the view model?

Reply:
Maybe not needed if it's hardcoded